extension/js/common/message-renderer.ts

```diff
@@ -430,8 +430,8 @@ export class MessageRenderer {
   };
 
   private getMessageInfo = async (fullMsg: GmailRes.GmailMsg): Promise<MessageInfo> => {
-    const sentDate = GmailParser.findHeader(fullMsg, 'date');
-    const sentDateStr = sentDate ? Str.fromDate(new Date(sentDate)).replace(' ', ' at ') : '';
+    const sentDate = $('div.gK span[title]').attr('title');
```

Comment:
let's also add check that this value is a valid date (for example, if Google will change it's markup in the future and `div.gK` will have another content).

and if it's value is invalid or empty, then we should have fallback by using value parsed from `fullMsg` (`GmailParser.findHeader(fullMsg, 'date');`). it can also be converted to user's timezone, so print info will show correct time for user.

Reply:
Hi @sosnovsky the `.toISOString()` seem not to be one we're looking for for a function that is aware of user's timezone and so `.toLocaleString()` be my option to use.

That being said this would be inappropriate now for this PR but would be interested to hear your feedback before I will place it as a new PR and close this one since we would no longer be needed to pull the date and time from Gmail web UI.

Reply:
Looks clean 👍 But need to check if it doesn't break anything, as several tests are failing now.

Reply:
Yes, Roma — I'll do it. I've come up with a different idea to improve them. Hopefully, it works. Be right back shortly.

Reply:
Hello @sosnovsky I ended up adding a fallback option as the original suggestion suggested. I think it would be the best option to go with.